File: 400/434 - Number of Segments in a String.c
int countSegments(char* s) {
    int ret = 0;
    
    while (*s) {
        if (*s == ' ') {
            s++;
            continue;
        }
        ++ret;
        
        while (*s && *s != ' ')
            ++s;
    }
    
    return ret;
}

File: csharp/434.number-of-segments-in-a-string.cs
/*
 * @lc app=leetcode id=434 lang=csharp
 *
 * [434] Number of Segments in a String
 */
public class Solution {
    public int CountSegments(string s) {
        int ret = 0;
        bool flag = false;
        foreach (char c in s) {
            if (c == ' ') {
                if (flag) {
                    ++ret;
                    flag = false;
                }
            } else {
                flag = true;
            }
        }
        if (flag) {
            ++ret;
        }
        return ret;
    }
}

